src/docfx/build/moniker/MonikersProvider.cs

```diff
@@ -3,23 +3,44 @@
 
 using System;
 using System.Collections.Generic;
+using System.IO;
 using System.Linq;
 
 namespace Microsoft.Docs.Build
 {
     internal class MonikersProvider
     {
         private readonly List<(Func<string, bool> glob, string monikerRange)> _rules = new List<(Func<string, bool>, string)>();
+        private readonly MonikerRangeParser _rangeParser;
+        private readonly Lazy<MonikerComparer> _monikerAscendingComparer;
+        private readonly Lazy<MonikerComparer> _monikerDescendingComparer;
 
-        public MonikersProvider(Config config)
+        public MonikerComparer AscendingComparer => _monikerAscendingComparer.Value;
+
+        public MonikerComparer DescendingComparer => _monikerDescendingComparer.Value;
+
+        public MonikersProvider(Docset docset, Config config)
         {
             foreach (var (key, monikerRange) in config.MonikerRange)
             {
                 _rules.Add((GlobUtility.CreateGlobMatcher(key), monikerRange));
             }
             _rules.Reverse();
+
+            var monikerDefinition = new MonikerDefinitionModel();
+            if (!string.IsNullOrEmpty(config.MonikerDefinition))
+            {
+                var path = docset.GetFileRestorePath(config.MonikerDefinition);
```

Comment:
It should be enough to just input path here, avoid exposing the entire Docset

Reply:
Taking a `Docset` as input is fine, `config` can be accessed from `docset` so you just need a `docset` as the input.